src/backend/executor/nodeShareInputScan.c

```diff
@@ -191,7 +191,7 @@ static void shareinput_release_callback(ResourceReleasePhase phase,
 
 static void shareinput_writer_notifyready(shareinput_Xslice_reference *ref);
 static void shareinput_reader_waitready(shareinput_Xslice_reference *ref);
-static void shareinput_reader_notifydone(shareinput_Xslice_reference *ref, int nconsumers);
+static void _shareinput_reader_notifydone(shareinput_Xslice_reference *ref, int nconsumers);
```

Comment:
Maybe use `shareinput_reader_notifydone_internal`? This seems different from current naming style here.